### Try Model: SVM classification

In [1]:
from Starts.startmodskl import *

local_kwargs {'nan_drop_col': False, 'data_path_1': '2_Titanic/data/train.csv', 'nan_zero': False, 'data_path_3': '2_Titanic/data/test.csv', 'nan_mean_neighbors': False, 'drop_obj_col': False, 'nan_mean': True, 'nan_drop_row': False, 'data_path_2': '2_Titanic/data/titanic_no_nan.csv'}


In [2]:
nonan_data.head()

Survived  Pclass  Sex   Age  Ticket     Fare  Title  FamilySize
0         0       3    1  49.0     574   0.0000      5           0
1         1       2    0   7.0     571  26.2500      4           2
2         0       3    1  33.0     378   8.6625      5           0
3         0       3    1  24.0     519  24.1500      5           2
4         1       1    0  30.0       0  86.5000      4           0

In [3]:
# Choose method Support Vector Machine method to classify data
classifier, y_true, y_pred = StartModSKL.classification_svm(nonan_data, 'Survived', kernel='rbf')

classifier


Accuracies:  [0.60504202 0.61344538 0.62288136]

Mean of accuracies:  0.6137895836300622

Standard Deviation:  0.007286945549735036


SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto', kernel='rbf',
  max_iter=-1, probability=False, random_state=0, shrinking=True,
  tol=0.001, verbose=False)

In [4]:
# Compare the predicted value with true_value
len(y_true), len(y_pred)

(179, 179)

In [5]:
confusion_matrix(y_true, y_pred)

array([[111,   4],
       [ 58,   6]])

### Result: 
#### the correct predicted result 117, 
#### the wrong predicted result 62

In [6]:
# Choose method Support Vector Machine method to classify data
classifier, y_true, y_pred = StartModSKL.classification_svm(nonan_data, 'Survived', kernel='linear')

classifier


Accuracies:  [0.78571429 0.80672269 0.77966102]

Mean of accuracies:  0.7906993305796896

Standard Deviation:  0.011596595155556414


SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto', kernel='linear',
  max_iter=-1, probability=False, random_state=0, shrinking=True,
  tol=0.001, verbose=False)

In [7]:
confusion_matrix(y_true, y_pred)

array([[102,  13],
       [ 22,  42]])

### Result: 
#### the correct predicted result 144, 
#### the wrong predicted result 35

In [8]:
# Choose method Support Vector Machine method (Kernel='sigmoid') to classify data
classifier, y_true, y_pred = StartModSKL.classification_svm(nonan_data, 'Survived', kernel='sigmoid')

classifier


Accuracies:  [0.6092437  0.6092437  0.61016949]

Mean of accuracies:  0.6095522954944689

Standard Deviation:  0.000436423498809574


SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto', kernel='sigmoid',
  max_iter=-1, probability=False, random_state=0, shrinking=True,
  tol=0.001, verbose=False)

In [10]:
len(y_true), len(y_pred)

(179, 179)

In [11]:
confusion_matrix(y_true, y_pred)

array([[115,   0],
       [ 64,   0]])

### Result: 
#### the correct predicted result 115, 
#### the wrong predicted result 64